### PR TITLE
Fix opam unable to find executables on systems where users belong to more than 32 groups when opam is built using musl libc

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Details of the issue in musl described in https://www.openwall.com/lists/musl/2021/07/03/1
Fixes https://github.com/ocaml/opam/issues/5373

### CODE REVIEW

**Comment by dra27** [comment]
I think there's a couple of bits in `configure.ac` from https://github.com/ocaml/opam/pull/4265/commits/5377b60d1df9f6dd2c42b5a386630fcc745271b4#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810 which also need reverting.

We should double-check that `Unix.access` doesn't regress either https://github.com/ocaml/opam/pull/4265/commits/d6faa8d45a3b06270863fb1c8d0848d7b45c6f36 or https://github.com/ocaml/opam/pull/4265/commits/6f756efc48da31e917a4a36f4c557b3e155d076b (this is probably just a case of checking the Open Group POSIX docs... I'll be _amazed_ if it regresses the first and quite surprised if it regresses the second!). It would be nice to check that the original bug being fixed in #4265, but that may be too tricky to set-up again.

**Comment by dra27** [comment]
Annoying, `Unix.access` checks `ruid` and `rgid` rather than `euid` and `egid` (which is what `exec` will do). Ideally we therefore want to use `faccessat` with `AT_EACCESS` instead, which should be exactly equivalent to the present check, but without needing the libacl hackery for Cygwin (and, in fairness, any other Unix system which chooses to grant permissions with ACLs instead). However, that would cause `opam` to require C stubs for all builds, and I'm not so sure about that :shrug:

**Comment by dra27** [comment]
Discussed today: we decided we're not worried about requiring stubs in opam-core for all platforms, so wrapping [`faccessat`](https://pubs.opengroup.org/onlinepubs/9699919799.2008edition/functions/faccessat.html) should be fine, which we can then use instead of `Unix.access`. Happy for either of us to get to this - either way, we get rid of the libacl bindings!

File: master_changes.md [hunk 1]
```diff
@@ -157,6 +157,7 @@ users)
   * Reduce allocations in OpamVersionCompare [#6144 @talex5]
   * Speedup OpamVersionCompare by 25% by removing the unused handling of epoch [#5518 @kit-ty-kate]
   * Fix error in `OpamSystem.transform_patch` - patches were only applied when debugging [#6182 @dra27 regression since #3449]
+  * Remove --with-acl option (reverts a Cygwin fix in #4265) [#5381 @kit-ty-kate - fix #5373]
```
Comment (rjbou) [inline master_changes.md:160]:
```suggestion
  * Remove `--with-acl` option from configure and its related C stubs(reverts a Cygwin fix in #4265) [#5381 @kit-ty-kate - fix #5373]
```

Comment (rjbou) [reply master_changes.md:160]: + update of API changes 
```
  * `OpamACL`: remove module
```

File: src/stubs/common/opamCommonStubs.mli [hunk 2]
```diff
@@ -1 +0,0 @@
-external check_executable : string -> unit = "opam_check_executable"
```
Comment (rjbou) [inline src/stubs/common/opamCommonStubs.mli]: These files were added in previous commit, should we squash this commit with 4e9b31230e7db1973407c35b772f02d947dc98d3 ?

File: src/core/opamStubsTypes.ml [hunk 3]
```diff
@@ -114,7 +114,7 @@ type win32_version_info = {
     (** Non-fixed string table. First field is a pair of Language and Codepage ID. *)
 }
 
-external check_executable : string -> unit = "opam_check_executable"
+external is_executable : string -> bool = "opam_is_executable"
```
Comment (rjbou) [inline src/core/opamStubsTypes.ml:117]: `check_executable` was added in this PR, is it possible to have a single commit that adds it as `is_executable` ?

File: src/core/opamCommonStubs.c [hunk 4]
```diff
@@ -1,15 +1,10 @@
 /**************************************************************************/
 /*                                                                        */
-/*                                 OCaml                                  */
+/*    Copyright 2024 Kate Deplaix                                         */
```
Comment (rjbou) [inline src/core/opamCommonStubs.c:3]: Same, it should be squashed

**Review by rjbou — COMMENTED** [review]
Some format remarks : commits to squash, etc.
Also, commit messages are not very explicit, an update with more information could help.

File: master_changes.md [hunk 5]
```diff
@@ -272,4 +272,5 @@ users)
   * `OpamHash`: export `compare_kind` [#5561 @rjbou]
   * `OpamFilename`: add `might_escape` to check if a path is escapable, ie contains `<sep>..<sep>` [#5561 @rjbou]
   * Add `OpamStd.Sys.getconf` [#5950 @kit-ty-kate]
+  * Fix opam unable to find executables on systems where users belong to more than 32 groups when opam is built using musl libc [#5381 @kit-ty-kate - fix #5373]
```
Comment (rjbou) [inline master_changes.md:277]:
```suggestion
  * `OpamStd.Sys.resolve_command`: Fix opam unable to find executables on systems where users belong to more than 32 groups when opam is built using musl libc [#5381 @kit-ty-kate - fix #5373]

**Review by rjbou — COMMENTED** [review]
Does `doc/index.html` need to be updated by the last commit (or others) ?

**Comment by kit-ty-kate** [comment]
> Does `doc/index.html` need to be updated by the last commit

done. Feel free to squash if you're fine with the two last fixup commits

**Review by rjbou — APPROVED** [review]
Thanks!

**Pull request merged**